optuna/integration/wandb.py

```diff
@@ -126,10 +155,33 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
         metrics = {name: value for name, value in zip(names, trial.values)}
         attributes = {"direction": [d.name for d in study.directions]}
 
-        wandb.config.update(attributes)
-        wandb.log({**trial.params, **metrics}, step=trial.number)
+        run_id = trial._user_attrs.get("run_id")
 
-    def _initialize_run(self) -> None:
-        """Initializes Weights & Biases run."""
+        if not (self._as_sweeps and run_id):
+            run = wandb.run or self._initialize_run()
+            step = trial.number if wandb.run else None
+            run.log({**trial.params, **metrics}, step=step)
```

Comment:
You're right. I should put `step = ...` before `run = ...`. The reason I'm not putting `step = trial.number` in all the runs, is because, everytime a new run is initialized, the result should be logged at `step = 0`, not at `step = trial.number`.